tests/models/test_factory.py

```diff
@@ -15,6 +15,8 @@
     "mixer_test_model",  # mlp_mixer.py
     "resmlp_test_model",
     "gmlp_test_model",
+    "resnet_test_model_sn",  # spectral normalization (SN)
+    "convnext_test_model_sn",
```

Comment:
Yes, this is unfortunately necessary because `self.u` and `self.v` for the SpectralNormalization will not have the right size if you change the input size.

Reply:
I am puzzled. Why does this code run without problems?
```
import numpy as np

import tfimm

model = tfimm.create_model("convnext_tiny", use_spec_norm=True)

y = model(np.zeros((2, 112, 112, 3)))
print(y.shape)

y = model(np.zeros((2, 224, 224, 3)))
print(y.shape)

y = model(np.zeros((3, 57, 57, 3)))
print(y.shape)
```

Reply:
The problem is when we are trying to transfer weights from one of the models above to the other, which is what we test for.

When I comment out the models from the exception (and we run the test), we get the following error [here](https://github.com/martinsbruveris/tensorflow-image-models/pull/70/checks#step:8:902):

```
ValueError: Cannot assign value to variable ' resnet_test_model_sn/layer1/0/conv1/v:0': Shape mismatch.The variable shape (1, 18, 18, 64), and the assigned value shape (1, 10, 10, 64) are incompatible.
```

Reply:
So, yes. We can create models of different input sizes, with spectral normalisation. But we cannot transfer weights between models of different input sizes.

Reply:
I understand the problem now. This is a weird one. Yes, setting the models as fixed input size works while at the same time it is not actually necessary to fix the input size. Also, I believe that changing the number of input channels will not work with the spectral normalisation...

Ok, we keep them in the fixed input size category for now.